scene/main/canvas_item.cpp

```diff
@@ -1683,3 +1683,6 @@ CanvasTexture::~CanvasTexture() {
 	ERR_FAIL_NULL(RenderingServer::get_singleton());
 	RS::get_singleton()->free(canvas_texture);
 }
+
+#define ERR_DRAW_GUARD \
```

Comment:
Does this work with this being at the end of the file? I would expect it at the start (after includes, by convention).

Reply:
I didn't know that was the convention. I will move it all the way up.

Reply:
Well it's not _just_ a convention ;)
```
$ cat main.cpp 
int main() {
        return SOME_VALUE;
}

#define SOME_VALUE 1
$ g++ main.cpp 
main.cpp: In function ‘int main()’:
main.cpp:2:16: error: ‘SOME_VALUE’ was not declared in this scope
    2 |         return SOME_VALUE;
      |                ^~~~~~~~~~
main.cpp:2:16: note: the macro ‘SOME_VALUE’ had not yet been defined
main.cpp:5: note: it was later defined here
    5 | #define SOME_VALUE 1
      |
```